src/jdk.jartool/share/classes/sun/tools/jar/FingerPrint.java

```diff
@@ -232,85 +242,73 @@ public int hashCode() {
         }
     }
 
-    private static final class ClassAttributes extends ClassVisitor {
-        private String name;
+    private static final class ClassAttributes implements Consumer<ClassElement> {
+        private final String name;
         private String outerClassName;
-        private String superName;
-        private int majorVersion;
-        private int access;
-        private boolean publicClass;
-        private boolean nestedClass;
+        private final String superName;
+        private final int majorVersion;
+        private final int access;
+        private final boolean publicClass;
+        private boolean maybeNestedClass;
```

Comment:
should be final since this is no longer updated.

Reply:
fixed, thanks.